extractor/src/main/java/org/schabi/newpipe/extractor/stream/StreamInfo.java

```diff
@@ -92,18 +96,20 @@ public static StreamInfo getInfo(StreamExtractor extractor) throws ExtractionExc
         return streamInfo;
     }
 
-    private static StreamInfo extractImportantData(StreamExtractor extractor) throws ExtractionException {
-        /* ---- important data, without the video can't be displayed goes here: ---- */
-        // if one of these is not available an exception is meant to be thrown directly
-        // into the frontend.
-
-        int serviceId = extractor.getServiceId();
-        String url = extractor.getUrl();
-        String originalUrl = extractor.getOriginalUrl();
-        StreamType streamType = extractor.getStreamType();
-        String id = extractor.getId();
-        String name = extractor.getName();
-        int ageLimit = extractor.getAgeLimit();
+    @Nonnull
+    private static StreamInfo extractImportantData(@Nonnull final StreamExtractor extractor)
+            throws ExtractionException {
+        /* ---- Important data, without it the stream can't be displayed: ---- */
+        // If one of these is not available an exception is meant to be thrown directly into
+        // NewPipe.
```

Comment:
In case of Piped or Skytube the exception is not thrown in NewPipe :P

Reply:
I'd say "the frontend will receive an exception directly"